pypit/arproc.py

```diff
@@ -2416,12 +2542,77 @@ def sub_overscan(frame, det):
     return frame
 
 
-def trim(frame, det):
-    dnum = settings.get_dnum(det)
-    for i in range(settings.spect[dnum]['numamplifiers']):
-        datasec = "datasec{0:02d}".format(i+1)
-        x0, x1 = settings.spect[dnum][datasec][0][0], settings.spect[dnum][datasec][0][1]
-        y0, y1 = settings.spect[dnum][datasec][1][0], settings.spect[dnum][datasec][1][1]
+def replace_columns(img, bad_cols, replace_with='mean'):
+    """ Replace bad columns with values from the neighbors
+
+    Parameters
+    ----------
+    img : ndarray
+    bad_cols: ndarray (bool, 1D, shape[1] of img)
+      True = bad column
+      False = ok column
+    replace_with : str, optional
+      Option for replacement
+       mean -- Use the mean of the closest left/right columns
+
+    Returns
+    -------
+    img2 : ndarray
+      Copy of the input image with the bad columns replaced
+    """
+    # Prep
+    img2 = img.copy()
+    # Find the starting/ends of the bad column sets
+    tmp = np.zeros(img.shape[1]).astype(int)
```

Comment:
Does anyone know if this is better than `np.zeros(img.shape[1], dtype=int)` ?  Probably marginal difference, but I usually try to create things with a given type using, e.g., `dtype=int` instead of creating the object and then converting the type.

Reply:
I believe they are equivalent.

Reply:
I'm not so sure... I think you are first creating the array and type, which then presumably gets dumped into a new array of type `int`, and the type of each value is converted. On my computer:

```
%timeit np.zeros(10000, dtype=int)
100000 loops, best of 3: 3.49 µs per loop

%timeit np.zeros(10000).astype(np.int)
100000 loops, best of 3: 8.1 µs per loop
```

possibly a good idea to keep this in mind for future, too...